src/renderer/flowtype-plugin.js

```diff
@@ -0,0 +1,116 @@
+var spawnSync = require('child_process').spawnSync;
+var flow = require('flow-bin');
+var merge = require('lodash.merge');
+
+var store = {
+  error: null,
+  flowOptions: [
+    'status',
+    '--color=always',
+  ],
+  options: {
+    warn: false,
+
+    formatter: function (errorCode, errorDetails) {
+      return 'Flow: ' + errorCode + '\n\n' + errorDetails;
+    },
+  },
+};
+
+
+function flowErrorCode(status) {
+  var error;
+  switch (status) {
+    /*
+    case 0:
+      error = null;
+      break;
+    */
+    case 1:
+      error = 'Server Initializing';
+      break;
+    case 2:
+      error = 'Type Error';
+      break;
+    case 3:
+      error = 'Out of Time';
+      break;
+    case 4:
+      error = 'Kill Error';
+      break;
+    case 6:
+      error = 'No Server Running';
+      break;
+    case 7:
+      error = 'Out of Retries';
+      break;
+    case 8:
+      error = 'Invalid Flowconfig';
+      break;
+    case 9:
+      error = 'Build Id Mismatch';
+      break;
+    case 10:
+      error = 'Input Error';
+      break;
+    case 11:
+      error = 'Lock Stolen';
+      break;
+    case 12:
+      error = 'Could Not Find Flowconfig';
+      break;
+    case 13:
+      error = 'Server Out of Date';
+      break;
+    case 14:
+      error = 'Server Client Directory Mismatch';
+      break;
+    case 15:
+      error = 'Out of Shared Memory';
+      break;
+  }
+
+  return error;
+}
+
+
+function checkFlowStatus(compiler, next) {
+  var res = spawnSync(flow, store.flowOptions);
```

Comment:
I think we should try to use `let` and `const` to show if a variable will change or not

Reply:
this file contains code from another repo, I don't want to change it too much because, once they update their npm package, we will be able to use that instead of this file.

Reply:
What is the other module? Seems simple enough, do we need to use it?

Reply:
this is a pretty decently maintained package: https://github.com/zhirzh/flow-babel-webpack-plugin

I'd like to be able to include other changes made by that developer if possible. I left an issue about updating the NPM package. If it looks abandoned after enough time then we should look at bringing this in-house. I expect to deprecate this file relatively soon though.

Reply:
but your right, if need be it is definitely simple